elephant/causality/granger.py

```diff
@@ -596,3 +776,163 @@ def conditional_granger(signals, max_order, information_criterion='aic'):
+        When given as a `float`, it is taken as frequency in Hz.
+        Default: None.
+    n_segments : int, optional
+        Number of segments. The length of segments is adjusted so that
+        overlapping segments cover the entire stretch of the given data. This
+        parameter is ignored if `len_segment` or `frequency_resolution` is
+        given.
+        Default: 1
+    len_segment : int, optional
+        Length of segments. This parameter is ignored if `frequency_resolution`
+        is given. If None, it will be determined from other parameters.
+        Default: None
+    frequency_resolution : pq.Quantity or float, optional
+        Desired frequency resolution of the obtained spectral Granger Causality
+        estimate in terms of the interval between adjacent frequency bins. When
+        given as a `float`, it is taken as frequency in Hz.
+        If None, it will be determined from other parameters.
+        Default: None
+    overlap : float, optional
+        Overlap between segments represented as a float number between 0 (no
+        overlap) and 1 (complete overlap).
+        Default: 0.5 (half-overlapped)
+    num_iterations : int
+        Number of iterations for algorithm to estimate spectral factorization.
+        Default: 300
+
+    Returns
+    -------
+    freqs : np.ndarray
+        Frequencies associated with the spectral Granger Causality estimate.
+    Causality
+        A `namedtuple` with the following attributes:
+            directional_causality_x_y : np.ndarray
+                Spectrally resolved Granger causality of X influence onto Y.
+
+            directional_causality_y_x : np.ndarray
+                Spectrally resolved Granger causality of Y influence onto X.
+
+            instantaneous_causality : np.ndarray
+                The remaining spectrally resolved channel interdependence not
+                accounted for by the directional causalities (e.g. shared input
+                to X and Y).
+
+            total_interdependence : np.ndarray
+                The sum of the former three metrics. It measures the dependence
+                of X and Y. If the total interdependence is positive, X and Y
+                are not independent.
+    """
+    if isinstance(signal_i, neo.core.AnalogSignal) and \
+            isinstance(signal_j, neo.core.AnalogSignal):
+        signals = signal_i.merge(signal_j)
+    elif isinstance(signal_i, np.ndarray) and isinstance(signal_j, np.ndarray):
+        signals = np.vstack([signal_i, signal_j])
+
+    # Calculate cross spectrum for signals
+    freqs, S = segmented_multitaper_cross_spectrum(
+        signals=signals, n_segments=n_segments, len_segment=len_segment,
+        frequency_resolution=frequency_resolution, overlap=overlap, fs=fs,
+        nw=nw, num_tapers=num_tapers, peak_resolution=peak_resolution,
+        return_onesided=False)
+
+    # Remove units attached by the multitaper_cross_spectrum
+    if isinstance(S, pq.Quantity):
```

Comment:
Why just ignore the units? Maybe you can ensure they are always rescaled to a certain value?